src/gasPriceOracle/oracle.ts

```diff
@@ -0,0 +1,98 @@
+import { BigNumber, providers } from "ethers";
+import { toBN } from "../utils";
+
+export type GasPriceEstimate = {
+  maxFeePerGas: BigNumber;
+  maxPriorityFeePerGas?: BigNumber;
+};
+
+type ValidatedFeeData = {
+  [idx: string]: BigNumber;
```

Comment:
nb. this is annoying, but seems necessary to perform a string-based lookup into a ValidatedFeeObject.

Reply:
Sorry, why is this necessary? What call fails?

Reply:
If I remove the line (as follows), tsc bails out on me:

```
  9 type ValidatedFeeData = {
 10   // [idx: string]: BigNumber;
 11   gasPrice: BigNumber;
 12   maxFeePerGas: BigNumber;
 13   maxPriorityFeePerGas: BigNumber;
 14 };
```

```
✖ Failed to compile
(typescript) Error: /home/pxrl/dev/across-protocol/sdk-v2/src/gasPriceOracle/oracle.ts(34,9): semantic error TS7053: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'ValidatedFeeData'.
  No index signature with a parameter of type 'string' was found on type 'ValidatedFeeData'.
```

The offending code is the `feeData[key]` lookup on line 32:

```
 29   try {
 30     const response: providers.FeeData = await provider.getFeeData();
 31     const expectedKeys: string[] = Object.keys(feeData);
 32     Object.entries(response).forEach(([key, value]) => {
 33       if (expectedKeys.includes(key) && BigNumber.isBigNumber(value) && value.gt(0)) {
 34         feeData[key] = toBN(value);
 35       }
 36     });
 37   }
```

So it seems like I am prevented from trying to lookup `feeData[key]` because the ValidatedFeeData type defines specific keys, and tsc can't quite recognise that I have already guaranteed that (indirectly) by validating that `key` exists in ValidatedFeeData.